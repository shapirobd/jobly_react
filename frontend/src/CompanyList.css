.CompanyList {
	width: 60vw;
	margin: 0 auto;
	text-align: center;
}

.CompanyList form {
	width: 100%;
}

.CompanyList input {
	box-sizing: border-box;
	border-radius: 5px;
	font-size: 20px;
	width: 100%;
	margin-top: 40px;
	padding: 14px;
	border: 1px solid rgb(179, 179, 179);
}
