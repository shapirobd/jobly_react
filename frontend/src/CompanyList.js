import React, { useEffect, useState } from "react";
import JoblyApi from "./api";
import Company from "./Company";
import "./CompanyList.css";

const CompanyList = () => {
	const [inputText, setInputText] = useState("");
	const [companies, setCompanies] = useState([]);

	useEffect(() => {
		const getCompanies = async (inputText) => {
			const resp = await JoblyApi.getCompanies(inputText);
			setCompanies(resp);
		};
		getCompanies(inputText);
	}, [inputText]);

	const handleChange = (e) => {
		const { value } = e.target;
		setInputText(value);
	};
	return (
		<div className="CompanyList">
			<form>
				<input
					type="text"
					value={inputText}
					onChange={handleChange}
					placeholder="Search for company name..."
				/>
			</form>
			{companies.map((c) => (
				<Company
					name={c.name}
					description={c.description}
					logoUrl={c.logoUrl}
					handle={c.handle}
					key={c.handle}
				/>
			))}
		</div>
	);
};

export default CompanyList;
