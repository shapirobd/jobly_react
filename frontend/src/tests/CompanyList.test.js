import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CompanyList from "../CompanyList";

test("smoke test", () => {
	render(
		<MemoryRouter>
			<CompanyList />
		</MemoryRouter>
	);
});

test("snapshot test", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<CompanyList />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
