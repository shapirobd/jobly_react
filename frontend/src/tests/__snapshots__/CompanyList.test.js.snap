// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div
    class="CompanyList"
  >
    <form>
      <input
        placeholder="Search for company name..."
        type="text"
        value=""
      />
    </form>
  </div>
</DocumentFragment>
`;
